### PR TITLE
Enable setup and installation on Balfrin (CSCS)

### LINKED ISSUE #41: Enable setup and installation on Balfrin (CSCS)
Currently the setup script and hence the installation fails on Balfrin. I will make a PR which expands the setup script to also work on Balfrin.

### PR DESCRIPTION
**Pull request by sadamov**
Fixes #41

### CODE REVIEW

**Comment by sadamov** [comment]
I made some cosmetic changes to the docs and upgraded the python version. As these changes are so minor I don't think we need a separate PR.

**Comment by cosunae** [comment]
thanks @sadamov for the contribution. 
In view of what was required to extend the installation to balfrin, I would like to propose a different mechanism to the installation procedure, based on the following thoughts:
* installation methods like if $machine == "XX"  are harder to maintained, as opposed to installation methods like conda install, spack install, that do not need to maintain specific instructions for various machines. 
* The global spack installations will not be supported in the future, therefore we should consider moving away from the use of
 source  /project/g110/spack/user/admin-daint/spack/share/spack/setup-env.sh
* I would avoid combining multiple package manager, in this case conda and spack. Since we use conda, do we need spack (see the following) ? 
* I think iconarray should not depend directly on eccodes, since it depends already on cfgrib, which will depend on eccodes. Therefore using two versions of eccodes (conda and spack) is not necessary and potentially conflicting. 
* The dependency on cosmo eccodes definitions can be managed by cloning: 
https://github.com/COSMO-ORG/eccodes-cosmo-resources
and setting the env variable GRIB_DEFINITION_PATH pointing to it. Since there is link, there is also probably no need to use spack. 
* With all of the above, reducing the installation to git clone of cosmo-eccodes-definitions and setting GRIB_DEFINITION_PATH, we should not need to specialize for each machine, and therefore we could aim a single instruction for all machines ? 

I hope this helps, I let the main reviewer @victoria-cherkas to decide if that makes sense and whether it belongs to this PR or somewhere else.

**Comment by victoria-cherkas** [comment]
Sorry I missed there was a PR for this [issue](https://github.com/C2SM/iconarray/issues/41), thanks @sadamov. 

@cosunae I agree these all make sense, and remove the dependency on spack which would be great. I would propose that we do that in this PR, I can help with this.

**Comment by victoria-cherkas** [comment]
Hope its OK, I added a refactor of the setup script with the changes outlined by @cosunae . While the re-gridding function points to fieldextra we would still need one if elif section to set the fieldextra path per machine, but elsewhere the script should now be machine agnostic.

File: doc/gettingstarted.rst [hunk 1]
```diff
@@ -20,53 +20,58 @@ First clone the iconarray repository and ``cd`` into its parent directory:
...
 
       conda activate iconarray
 
 
-Alternatively if you are adding iconarray to your **existing conda environment**, carry out these two steps:
+Alternatively if you are adding iconarray to your **existing conda environment**,
+carry out these two steps:
+
+1. Update your existing conda environment by executing this command.
+2. It will install iconarray and all its missing dependencies into your existing conda environment:
```
Comment (victoria-cherkas) [inline doc/gettingstarted.rst:39]: I'm not sure this change is needed, a single step is now broken into two steps?

File: doc/gettingstarted.rst [hunk 2]
```diff
@@ -20,53 +20,58 @@ First clone the iconarray repository and ``cd`` into its parent directory:
...
 
-1. Update your existing conda environment by executing this command. It will install iconarray and all its missing dependencies into your existing conda environment:
    .. code::
-      
+
       conda env update -n {YOUR_ENVIRONMENT} -f env/environment.yml
 
-2. You can then re-export an updated ``environment.yml`` file of your environment:
+3. You can then re-export an updated ``environment.yml`` file of your environment:
```
Comment (victoria-cherkas) [inline doc/gettingstarted.rst:45]: And I would keep this as the second step

**Comment by sadamov** [comment]
Oh these are great suggestions, much easier setup now! I quickly tested it on Balfrin and Tsa and it works for me out of the box. 🚀

File: doc/gettingstarted.rst (hunk 1 above)
Comment (sadamov) [reply doc/gettingstarted.rst]: it's not needed, you are right

Comment (sadamov) [reply doc/gettingstarted.rst]: Yes I saw, sorry forgot to comment earlier

**Comment by victoria-cherkas** [comment]
launch jenkins

**Comment by victoria-cherkas** [comment]
Tests failing, because we are now using GRIB definition files from later eccodes version. I believe this newer eccodes version is required by current version of cfgrib. Looking into it

**Comment by victoria-cherkas** [comment]
launch jenkins

**Comment by victoria-cherkas** [comment]
launch jenkins

**Comment by victoria-cherkas** [comment]
Jenkins tests passing with eccodes 2.25.0 and eccodes-cosmo-resources v2.25.0.1

**Review by victoria-cherkas — APPROVED** [review]

**Pull request merged**